### PR TITLE
Trash API endpoints parity & unit test coverage

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: boxsdk/client/client.py [hunk 1]
```diff
@@ -391,3 +393,33 @@ def get_url(self, endpoint, *args):
...
+            An iterator of the entries in the trash
+        :rtype:
+            :class:`BoxObjectCollection`
+        """
+        return LimitOffsetBasedObjectCollection(
+            session=self._session,
+            url='{0}/folders/trash/items'.format(API.BASE_API_URL),
+            limit=limit,
+            offset=offset,
+            return_full_pages=False
```
Comment (potrebic) [inline boxsdk/client/client.py]: trailing comma

File: boxsdk/client/client.py [hunk 2]
```diff
@@ -391,3 +393,33 @@ def get_url(self, endpoint, *args):
         """
         # pylint:disable=no-self-use
         return self._session.get_url(endpoint, *args)
+
+    def get_trashed_items(self, offset=None, limit=None, fields=None):
+        """
+        Get the entries in the trashed items using limit-offset paging.
```
Comment (potrebic) [inline boxsdk/client/client.py]: Our online docs read nicer, which would translate to the following here:
```
Using limi-offset paging, get the files, folders and web links that are in the user's trash
```

File: boxsdk/object/item.py [hunk 3]
```diff
@@ -322,3 +323,71 @@ def delete(self, params=None, etag=None):
...
+        :type fields:
+            `Iterable` of `unicode`
+        """
+        url = self.get_url()
+        body = {
+            'name': name,
+            'parent': {
+                'id': parent_id
+            }
+        }
```
Comment (potrebic) [inline boxsdk/object/item.py]: 2 missing trailing commas - L371 and L372

File: boxsdk/object/item.py [hunk 4]
```diff
@@ -322,3 +323,71 @@ def delete(self, params=None, etag=None):
...
+            session=self._session,
+            object_id=response['id'],
+            response_object=response,
+        )
+
+
+    def permanently_delete_item(self):
+        """
+        Permanently delete an item that is in the trash. The item will no longer exist in Box.
+
```
Comment (potrebic) [inline boxsdk/object/item.py]: remove blank line

File: boxsdk/object/trash.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# coding: utf-8
+
+from __future__ import unicode_literals, absolute_import
+
+from .base_endpoint import BaseEndpoint
+
+
+class Trash(BaseEndpoint):
+    """Box API endpoint for performing trash related actions in Box."""
```
Comment (potrebic) [inline boxsdk/object/trash.py]: Is this class used somewhere?

File: boxsdk/client/client.py [hunk 6]
```diff
@@ -15,6 +15,8 @@
 from ..object.group_membership import GroupMembership
 from ..util.shared_link import get_shared_link_header
 from ..util.translator import Translator
+from ..pagination.marker_based_object_collection import MarkerBasedObjectCollection
```
Comment (potrebic) [inline boxsdk/client/client.py]: I don't see any uses of `MarkerBasedObjectCollection` in the diffs? Is this import needed?

File: boxsdk/object/item.py [hunk 7]
```diff
@@ -7,6 +7,7 @@
 from .base_object import BaseObject
 from boxsdk.config import API
 from boxsdk.exception import BoxAPIException
+from ..pagination.marker_based_object_collection import MarkerBasedObjectCollection
```
Comment (potrebic) [inline boxsdk/object/item.py]: Is this import needed?

File: boxsdk/object/item.py [hunk 8]
```diff
@@ -322,3 +323,71 @@ def delete(self, params=None, etag=None):
         """
         headers = {'If-Match': etag} if etag is not None else None
         return super(Item, self).delete(params, headers)
+
+
+    def get_from_trash(self, fields=None):
```
Comment (potrebic) [inline boxsdk/object/item.py]: Should this and the calls below have the `@api_call` decorator?

File: boxsdk/object/item.py [hunk 9]
```diff
@@ -322,3 +323,71 @@ def delete(self, params=None, etag=None):
...
+        """
+        Get item from trash.
```
Comment (potrebic) [inline boxsdk/object/item.py]: This wording feels confusing. This call gets information. See the inherited get() call which says:
```
        Get information about the object, specified by fields. If fields is None, return the default fields.
```

File: boxsdk/object/item.py [hunk 10]
```diff
@@ -322,3 +323,71 @@ def delete(self, params=None, etag=None):
...
+        box_response = self._session.get(url, params=params)
+        response = box_response.json()
+        return self.__class__(
+            session=self._session,
+            object_id=response['id'],
+            response_object=response,
+        )
+
+
+    def restore_from_trash(self, name=None, parent_id=None, fields=None):
```
Comment (potrebic) [inline boxsdk/object/item.py]: Yippie!!! I've been wanting restore!

File: boxsdk/pagination/page.py [hunk 11]
```diff
@@ -38,7 +39,7 @@ def _translator(self):
 
         :rtype:   :class:`Translator`
         """
-        return self._session.translator
+        return Translator()
```
Comment (potrebic) [inline boxsdk/pagination/page.py]: I'm not sure what this change is doing / how it's related to the PR? Could you explain in the PR comment?

**Pull request closed without merging**